x/nexus/keeper/general_message.go

```diff
@@ -10,13 +10,23 @@ import (
 	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
 )
 
-func getMessageKey(id exported.MessageID) key.Key {
+func getMessageKey(id exported.MessageID, status exported.GeneralMessage_Status) key.Key {
 	return generalMessagePrefix.
 		Append(key.From(id.Chain)).
+		Append(key.From(status)).
 		Append(key.FromStr(id.ID))
 }
 
-// SetNewMessage sets the given general message
+// GetGeneralMessageID generates a unique general message ID
+func (k Keeper) GetGeneralMessageID(ctx sdk.Context, sourceTxID string, sourceChain exported.ChainName) string {
```

Comment:
is tx id enough? there might be multiple msgs per tx, no?

Reply:
Since we append an incrementing nonce, it's ok if there are multiple msgs per tx